src/todo.js

```diff
@@ -0,0 +1,143 @@
+      localStorage.setItem('data', JSON.stringify(data));
+    }
+  });
+  checkbox.setAttribute('type', 'checkbox');
+  checkbox.checked = todos.completed;
+  const todoField = document.createElement('input');
+  todoField.className = 'todo-edt-input';
+  todoField.setAttribute('type', 'text');
+  todoField.addEventListener('input', () => {
+    if (todoField.value !== '') {
+      const data = JSON.parse(localStorage.getItem('data'));
+      const indx = todos.index;
+      const action = data.filter((todo) => indx === todo.index);
+      action[0].description = todoField.value;
+      data.splice(indx - 1, 1, action[0]);
+      localStorage.setItem('data', JSON.stringify(data));
+    }
+  });
+  const iconSpan = document.createElement('span');
+  iconSpan.className = 'symbol-option-cont';
+  iconSpan.innerHTML = '&#128465;';
+  iconSpan.addEventListener('click', () => {
+    const data = JSON.parse(localStorage.getItem('data'));
+    const indx = todos.index;
+    data.splice(indx - 1, 1);
+    let index = 1;
+    data.forEach((el) => {
+      el.index = index;
+      index += 1;
+    });
+    localStorage.setItem('data', JSON.stringify(data));
+    const itemslist = document.querySelector('.cont-todos');
+    itemslist.innerHTML = null;
+
+    data.forEach((todos) => {
+      const itemList = listItm(todos);
+      itemslist.appendChild(itemList);
+    });
+  });
+  itemlist.className = 'todo';
+  if (todos.completed === true) {
+    todoField.style.textDecoration = 'line-through';
+  }
+  itemlist.appendChild(checkbox);
+  todoField.value = todos.description;
+  itemlist.appendChild(todoField);
+  itemlist.appendChild(iconSpan);
+
+  return itemlist;
+}
+
+export default class AddTodo {
+  static populateTodos() {
+    if (localStorage.getItem('data')) {
+      const todosloc = JSON.parse(localStorage.getItem('data'));
+
+      const itemslist = document.querySelector('.cont-todos');
+      todosloc.forEach((todos) => {
+        const listItem = listItm(todos);
+        itemslist.appendChild(listItem);
+      });
+    }
+  }
+
+  static add() {
+    const fieldTodo = document.querySelector('.new-todo-field');
+    const listItems = document.querySelector('.cont-todos');
+    fieldTodo.addEventListener('keypress', (e) => {
+      if (e.key === 'Enter') {
+        if (localStorage.getItem('data') === null && fieldTodo.value.length > 0) {
+          const todosList = [];
+          const todos = {
+            description:
+             fieldTodo.value,
+            completed: false,
+            index: todosList.length + 1,
+          };
+          todosList.push(todos);
+          localStorage.setItem('data', JSON.stringify(todosList));
+          const itmList = listItm(todos);
+          listItems.appendChild(itmList);
+          fieldTodo.value = '';
+        } else if (localStorage.getItem('data') !== null && fieldTodo.value.length > 0) {
+          const data = JSON.parse(localStorage.getItem('data'));
```

Comment:
According to the DRY principle, it will be nice if you can have a function to perform this task

Reply:
Yes, that is true I have made the changes